### ISSUE TITLE
Support TS Nightlies (2.x)

### ISSUE THREAD

**Issue by JKillian** [Microsoft/tslint-microsoft-contrib#91]
I'm considering adding tslint-microsoft-contrib as a dev-dependency of tslint which we'll use in tests to make sure custom rules are working correctly. However, we'd need things to work with TS nightly releases because we have a branch of TSLint that supports them. Any chance that could happen? Also, it would allow people who build with TS nightlies to use these rules.

**Comment by HamletDRC** [comment]
Sure. Sounds great. 

What work is needed on my side? 

We have a branch called "releases" that is just our npm package: https://github.com/Microsoft/tslint-microsoft-contrib/tree/releases

We update it every time we update the code (it is not nightly, but instead per-commit)

**Comment by JKillian** [comment]
@HamletDRC Actually, I ended up just writing one custom rule by myself that always creates a failure for our testing purposes, but I still think it would be neat for these rules to support the nightly version of TypeScript!

I think the primary change is just adding code [to this function](https://github.com/Microsoft/tslint-microsoft-contrib/blob/284f41775008f32b3a76b80fceace1127c935530/src/utils/SyntaxKind.ts#L1171-L1184) so it doesn't throw an error if the TypeScript version is above 1.7.x.

You actually shouldn't have to do this manual SyntaxKind caching for versions of TypeScript >= 1.7 because they're not typed as `const enum` anymore in typing files. See https://github.com/Microsoft/TypeScript/issues/5219

**Comment by HamletDRC** [comment]
sounds like tslint took care of this on their side

**Comment by JKillian** [comment]
Actually, my comment still stands because of this code:

```ts
    export function current(): SyntaxKind.SyntaxKind {
        if (cachedKinds == null) {
            if (/1\.5\..*/.test(ts.version)) {
                cachedKinds = SyntaxKind_1_5;
            } else if (/1\.6\..*/.test(ts.version)) {
                cachedKinds = SyntaxKind_1_6;
            } else if (/1\.7\..*/.test(ts.version)) {
                cachedKinds = SyntaxKind_1_7;
            } else if (/1\.8\..*/.test(ts.version)) {
                cachedKinds = SyntaxKind_1_8;
            } else {
                throw new Error('Unsupported TypeScript version: ' + ts.version);
            }
        }
        return cachedKinds;
    }
```

If you're using a nightly version of TypeScript, it's on version "1.9", so an error will be thrown. In fact, syntax kind caching shouldn't even be needed in versions 1.8+ because it's no longer an enum, so you can just use/return `ts.SyntaxKind` directly.

**Comment by blakeembrey** [comment]
Just run into this myself, will need to remove these TSLint rules until it's fixed.

**Comment by HamletDRC** [comment]
@blakeembrey What exactly is specified in your package.json, which rules are failing, and what errors do you see? I just tested with the latest releases of grunt-tslint, tslint, and TypeScript and there were no failures.

**Comment by blakeembrey** [comment]
I'm using nightlies, as the original issues and @JKillian mentioned.

**Comment by HamletDRC** [comment]
@blakeembrey Can you please post your package.json file or describe exactly which dependencies you are using? 

I do no receive any errors when I use any of these permutations: 
* Typescript stable + tslint nightly + tslint-microsoft-contrib nightly
* TypeScript nightly + tslint nightly + tslint-microsoft-contrib nightly 

What exactly are your dependencies set to? 

Also, which rules fail and what are the failures?

**Comment by HamletDRC** [comment]
@blakeembrey I assume you're seeing the error "TypeScript version 1.9 not supported". That issue is fixed and available on our "releases" branch. You can point your npm dependency to that branch in order to see the fix. 

If that was not the error you were receiving, then please log another issue. 

Thanks.

**Comment by blakeembrey** [comment]
Thanks, I wasn't aware there was a `tslint-microsft-contrib@next`.

**Comment by tp** [comment]
@HamletDRC Actually TypeScript nightlies recently switched from 1.9 to 2.0, so this is giving errors again.

Can 2.0 support be added [here](https://github.com/Microsoft/tslint-microsoft-contrib/blob/1590951ab1516200dd418219896767f46ecde514/src/utils/SyntaxKind.ts#L1791)?

**Comment by HamletDRC** [comment]
@tp I added support for 2.0 and pushed to the release branch.
